From ecd84fac48a56f1ebc984f1ee0d9dff30260c961 Mon Sep 17 00:00:00 2001
From: Alexander Larsson <alexl@redhat.com>
Date: Thu, 27 Sep 2012 17:24:31 +0000
Subject: iconview: Render focus even in non-selected items

This is needed for the SELECTION_NONE mode where nothing is ever
selected, but its also needed for CTRL-<key> keynav that moves the
focus without changing the selection.

https://bugzilla.gnome.org/show_bug.cgi?id=684984
---
diff --git a/gtk/gtkiconview.c b/gtk/gtkiconview.c
index a1df572..7eb50ae 100644
--- a/gtk/gtkiconview.c
+++ b/gtk/gtkiconview.c
@@ -3042,14 +3042,12 @@ gtk_icon_view_paint_item (GtkIconView     *icon_view,
 
   state &= ~(GTK_STATE_FLAG_SELECTED | GTK_STATE_FLAG_PRELIGHT);
 
+  if ((state & GTK_STATE_FLAG_FOCUSED) &&
+      item == icon_view->priv->cursor_item)
+    flags |= GTK_CELL_RENDERER_FOCUSED;
+
   if (item->selected)
     {
-      if ((state & GTK_STATE_FLAG_FOCUSED) &&
-          item == icon_view->priv->cursor_item)
-        {
-          flags |= GTK_CELL_RENDERER_FOCUSED;
-        }
-
       state |= GTK_STATE_FLAG_SELECTED;
       flags |= GTK_CELL_RENDERER_SELECTED;
     }
--
cgit v0.9.0.2
